cmd/teleirc.go

```diff
@@ -27,13 +27,20 @@ func main() {
 	flag.Parse()
 
 	settings, err := internal.LoadConfig(path)
+
 	if err != nil {
 		fmt.Println(err)
-	} else {
-		startTelegram()
-		client := irc.NewClient(settings.IRC)
-		if err := client.StartBot(); err != nil {
-			fmt.Println(err)
-		}
+		return
+	}
+
+	startTelegram()
+	client := irc.NewClient(settings.IRC)
+
+	ircChan := make(chan error)
+	go client.StartBot(ircChan)
+
+	select {
+	case err := <-ircChan:
```

Comment:
Are cases the best way to handle this? Are there any docs on that?

Reply:
Sure, [here's](https://gobyexample.com/non-blocking-channel-operations) an example from gobyexample.com. Pretty standard, allows you to block until one of the two channels returns and then executes the respective case statement.

Reply:
In this case, of course, there's only one until the Telegram bot has a channel for sending messages as well

Reply:
This use case is what `select` is made for, it's not the same as a switch statement

Reply:
@Zedjones thanks for the example. I'll take a look at this info when I can :+1: